esp/esp/users/controllers/usersearch.py

```diff
@@ -104,6 +107,29 @@ def query_from_criteria(self, user_type, criteria):
                 Q_include &= Q(studentregistration__section__parent_class__id__in=clsid,
                                studentregistration__relationship__name__in=student_verbs) & nest_Q(StudentRegistration.is_valid_qobject(), 'studentregistration')
 
+            if 'class_times' in criteria:
+                class_times = criteria['class_times']
+                if 'regtypes' in criteria:
+                    student_verbs = criteria['regtypes']
+                else:
+                    student_verbs = ['Enrolled']
+                times_filter = reduce(operator.or_, (Q(studentregistration__section__meeting_times__in=time, studentregistration__relationship__name__in=student_verbs)
+                                                     & nest_Q(StudentRegistration.is_valid_qobject(), 'studentregistration') for time in class_times))
+                Q_include &= times_filter
+                self.updated = True
+
+            if 'teaching_times' in criteria:
+                teaching_times = criteria['teaching_times']
+                times_filter = reduce(operator.or_, (Q(classsubject__sections__meeting_times__in=time) for time in teaching_times))
```

Comment:
Maybe I'm overlooking something here, but couldn't you just do `classsubject__sections__meeting_times__in=teaching_times`?

Reply:
Also, I think it should be `meeting_times__id` since the `teaching_times` are event ids?

Reply:
OK, so at first I was like, "wait, how is this even working?". I tried running this line by myself using a test list of event integer ids, and it kept erroring (because `time` wasn't iterable). However, what I discovered is that `time` here is actually a string, so what happens is it treats the string as an iterable, looking for events with an id that matches each character of the string. This works fine if your events all have single digit ids, but once you get into larger ids, you're then filtering inadvertently to the single digit ids that make up the desired multidigit value. My two comments above should fix that. You'll want to do that here, above, and below.